File: EducationalPlatform.Domain/Entities/Course.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EducationalPlatform.Domain.Entities
{
    public class Course
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; }

        [MaxLength(2000)]
        public string Description { get; set; }

        [Required]
        [ForeignKey(nameof(Instructor))]
        public Guid InstructorId { get; set; }

        public int? EstimatedDurationHours { get; set; }

        [Required]
        public bool IsActive { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual User Instructor { get; set; }
        public virtual ICollection<Lesson> Lessons { get; set; }
        public virtual ICollection<Enrollment> Enrollments { get; set; }
    }
}

File: frontend/src/types/course.ts
export interface CourseDto {
  id: string;
  title: string;
  description: string;
  instructorId: string;
  estimatedDurationHours: number;
  isActive: boolean;
}

export interface CreateCourseDto {
  title: string;
  description: string;
  instructorId: string;
  estimatedDurationHours: number;
  isActive: boolean;
}

export interface UpdateCourseDto {
  title: string;
  description: string;
  estimatedDurationHours: number;
  isActive: boolean;
}
